networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/libp2p/LibP2PNetwork.java

```diff
@@ -153,6 +160,19 @@ public LibP2PNetwork(
             });
   }
 
+  public Multiaddr composeAdvertisedAddr(NetworkConfig config) throws UnknownHostException {
+    String ip;
+    if (!config.getAdvertisedIp().trim().isEmpty()) {
+      ip = config.getAdvertisedIp();
```

Comment:
We probably also want to check that this at least looks like a valid ip: `InetAddresses.isInetAddress(config.getAdvertisedIp())`